pkg/assisted/agentclusterinstall_test.go

```diff
@@ -0,0 +1,1180 @@
+			expectError: "",
+		},
+		{
+			ip:          "notanip",
+			expectError: "agentclusterinstall ingressVIP incorrectly formatted",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithIngressVip(testCase.ip)
+		assert.Equal(t, testCase.expectError, testBuilder.errorMsg)
+
+		if testCase.expectError == "" {
+			assert.Equal(t, testCase.ip, testBuilder.Definition.Spec.IngressVIP)
+		}
+	}
+}
+func TestWithAdditionalIngressVip(t *testing.T) {
+	testCases := []struct {
+		ip          string
+		expectError string
+	}{
+		{
+			ip:          "192.168.1.10",
+			expectError: "",
+		},
+		{
+			ip:          "fd2e::10",
+			expectError: "",
+		},
+		{
+			ip:          "notanip",
+			expectError: "agentclusterinstall ingressVIP incorrectly formatted",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithAdditionalIngressVip(testCase.ip)
+		assert.Equal(t, testCase.expectError, testBuilder.errorMsg)
+
+		if testCase.expectError == "" {
+			assert.Contains(t, testBuilder.Definition.Spec.IngressVIPs, testCase.ip)
+		}
+	}
+}
+func TestWithUserManagedNetworking(t *testing.T) {
+	testCases := []struct {
+		umn bool
+	}{
+		{
+			umn: true,
+		},
+		{
+			umn: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithUserManagedNetworking(testCase.umn)
+
+		assert.Equal(t, testCase.umn, *testBuilder.Definition.Spec.Networking.UserManagedNetworking)
+	}
+}
+func TestWithPlatformType(t *testing.T) {
+	testCases := []struct {
+		platform hiveextV1Beta1.PlatformType
+	}{
+		{
+			platform: "BareMetal",
+		},
+		{
+			platform: "None",
+		},
+		{
+			platform: "VSphere",
+		},
+		{
+			platform: "Nutanix",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithPlatformType(testCase.platform)
+
+		assert.Equal(t, testCase.platform, testBuilder.Definition.Spec.PlatformType)
+	}
+}
+func TestWithControlPlaneAgents(t *testing.T) {
+	testCases := []struct {
+		count         int
+		expectedError string
+	}{
+		{
+			count:         3,
+			expectedError: "",
+		},
+		{
+			count:         1,
+			expectedError: "",
+		},
+		{
+			count:         0,
+			expectedError: "agentclusterinstall controlplane agents must be greater than 0",
+		},
+		{
+			count:         -1,
+			expectedError: "agentclusterinstall controlplane agents must be greater than 0",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithControlPlaneAgents(testCase.count)
+
+		assert.Equal(t, testCase.expectedError, testBuilder.errorMsg)
+		assert.Equal(t, testCase.count, testBuilder.Definition.Spec.ProvisionRequirements.ControlPlaneAgents)
+	}
+}
+
+func TestWithImageSet(t *testing.T) {
+	testCases := []struct {
+		image         string
+		expectedError string
+	}{
+		{
+			image:         "4.16",
+			expectedError: "",
+		},
```

Comment:
One more test with something that raises an error?

Reply:
Currently this method doesn't return any errors. I will see if there is a validation that can be added before. It was allowed to be empty in the past

Reply:
No, it looks like an empty imageset reference can still be passed in